### PR TITLE
Fix `expm(complex matrix)`

### PR DESCRIPTION
**Pull request by ev-br**
closes https://github.com/cupy/cupy/issues/8205

### CODE REVIEW

File: tests/cupyx_tests/scipy_tests/linalg_tests/test_matfuncs.py [hunk 1]
```diff
@@ -103,3 +103,8 @@ def test_2x2_input(self, xp, scp):
     def test_nx2x2_input(self, xp, scp, a):
         a = xp.asarray(a)
         return scp.linalg.expm(a)
+
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_complex(self, xp, scp):
+        a = xp.eye(2, dtype=xp.complex128)
+        return scp.linalg.exmp(a)
```
Comment (leofang) [inline tests/cupyx_tests/scipy_tests/linalg_tests/test_matfuncs.py:110]: misspelling 
```suggestion
        return scp.linalg.expm(a)
```

File: tests/cupyx_tests/scipy_tests/linalg_tests/test_matfuncs.py [hunk 2]
```diff
@@ -103,3 +103,8 @@ def test_2x2_input(self, xp, scp):
     def test_nx2x2_input(self, xp, scp, a):
         a = xp.asarray(a)
         return scp.linalg.expm(a)
+
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_complex(self, xp, scp):
```
Comment (leofang) [inline tests/cupyx_tests/scipy_tests/linalg_tests/test_matfuncs.py:108]: I am a bit confused, shouldn't we just have one test marked with `@testing.for_all_dtypes()` and we'd cover all spectrum? Seems like we don't have good dtype test coverage for `expm` even after this PR (ex: float16float32/float64/int/...)

**Review by leofang — CHANGES_REQUESTED** [review]

File: tests/cupyx_tests/scipy_tests/linalg_tests/test_matfuncs.py (hunk 2 above)
Comment (ev-br) [reply tests/cupyx_tests/scipy_tests/linalg_tests/test_matfuncs.py]: Fair enough. Updated to the suggestion.

File: tests/cupyx_tests/scipy_tests/linalg_tests/test_matfuncs.py [hunk 3]
```diff
@@ -103,3 +103,9 @@ def test_2x2_input(self, xp, scp):
     def test_nx2x2_input(self, xp, scp, a):
         a = xp.asarray(a)
         return scp.linalg.expm(a)
+
+    @testing.for_all_dtypes(no_bool=True, no_float16=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
```
Comment (ev-br) [inline tests/cupyx_tests/scipy_tests/linalg_tests/test_matfuncs.py:108]:
```suggestion
    @testing.numpy_cupy_allclose(scipy_name='scp', contiguous_check=False)
```

**Comment by ev-br** [comment]
The last commit relaxes the contiguity check for the new test, to address the CI failure. This parallels other expm tests, e.g. https://github.com/cupy/cupy/pull/8206/files#diff-8f1d8978c473f8ee597a50b45b106a148aec0ea14099c0d7c43318793ad3497cL91

**Review by takagi — APPROVED** [review]
LGTM!

**Comment by takagi** [comment]
Thanks, @ev-br!

**Comment by leofang** [comment]
@takagi Would it be OK to backport this PR as a bug-fix? This is a blocker for quantum computing users, where it's expected for `expm` to work for complex matrices.

**Comment by takagi** [comment]
Sure.

**Pull request merged**